### PR TITLE
Add python multi dimensional slice indexing

### PR DESCRIPTION
**Pull request by blowekamp**

### CODE REVIEW

File: Wrapping/Python/Python.i [hunk 1]
```diff
@@ -555,18 +554,15 @@
...
                 size = img.GetSize();
 
                 # set the slice dimension size to 0
-                size = size[:slice_dim]+(0,)+size[slice_dim+1:]
+                for i in slice_dims:
+                  size = tuple(size[:i]) + (0,) + tuple(size[i+1:])
 
-                # reduce the 3D image to a 2D
+                print("extract size: ", size)
+                # Redfuce the dimension of the image
```
Comment (dave3d) [inline Wrapping/Python/Python.i:565]: 'Reduce'

File: Wrapping/Python/Python.i [hunk 2]
```diff
@@ -482,8 +482,10 @@
             of the image it will be padded with the defaults slice
             ":".
 
-            A 2D image can be extracted from a 3D image by providing
-            one argument being an integer instead of a slice."""
+            When an index element is an integer, that dimension is
+            colapsed extracting an image with reduced dimensionality.
```
Comment (dave3d) [inline Wrapping/Python/Python.i:486]: 'collapsed,'

**Review by dave3d — COMMENTED** [review]
Couple of typos

File: Testing/Unit/Python/sitkImageIndexingTest.py [hunk 3]
```diff
@@ -248,6 +258,42 @@ def test_4d(self):
          self.assertImageNDArrayEquals(img[-1:-4:-1,:,:,:], nda[:,:,:,-1:-4:-1])
 
 
+    def test_5d(self):
+         """testing __getitem__ for 5D image"""
+
+         # Check if we have 4D Image support
```
Comment (zivy) [inline Testing/Unit/Python/sitkImageIndexingTest.py:264]: "Check if we have 5D..."

**Review by zivy — APPROVED** [review]

File: Wrapping/Python/Python.i (hunk 1 above)
Comment (blowekamp) [reply Wrapping/Python/Python.i:565]: That print statement shouldn't be their either.

**Review by dave3d — APPROVED** [review]

**Pull request merged**